File: src/TaskOTime/TaskOTime.ViewModel/ViewModels/TimeEntryEditRequestEventArgs.vb
Imports System

Namespace ViewModels
    Public Class TimeEntryEditRequestEventArgs
        Inherits EventArgs

        Public Sub New(entryTime As DateTime, title As String, description As String, completeRunningTask As Boolean, saveAction As Action(Of DateTime, String, String, Boolean))
            Me.EntryTime = entryTime
            Me.Title = title
            Me.Description = description
            Me.CompleteRunningTask = completeRunningTask
            Me.SaveAction = saveAction
        End Sub

        Public Property EntryTime As DateTime

        Public Property Title As String

        Public Property Description As String

        Public Property CompleteRunningTask As Boolean

        Public ReadOnly Property SaveAction As Action(Of DateTime, String, String, Boolean)
    End Class
End Namespace

File: src/TaskOTime/TaskOTime.ViewModel/ViewModels/TimeEntryEditRequestEventArgs.cs
using System;

namespace TaskOTime.ViewModel.ViewModels
{
    public class TimeEntryEditRequestEventArgs : EventArgs
    {

        public TimeEntryEditRequestEventArgs(DateTime entryTime, string title, string description, bool completeRunningTask, Action<DateTime, string, string, bool> saveAction)
        {
            EntryTime = entryTime;
            Title = title;
            Description = description;
            CompleteRunningTask = completeRunningTask;
            SaveAction = saveAction;
        }

        public DateTime EntryTime { get; set; }

        public string Title { get; set; }

        public string Description { get; set; }

        public bool CompleteRunningTask { get; set; }

        public Action<DateTime, string, string, bool> SaveAction { get; private set; }
    }
}